workflow-templates/pre-commit/.github/workflow/pre-commit.yaml

```diff
@@ -0,0 +1,94 @@
+name: pre-commit
+
+on:
+  pull_request:
+    types: [opened, reopened, synchronize]
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: read
+
+jobs:
+  pre-commit-check:
+    runs-on: ubuntu-latest
+    steps:
+      # fetch-depth: 0 is required for committed to work properly
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Generate commit refs for push to main branch
+        if: github.event_name == 'push'
+        run: |
+          echo "from_ref=${{ github.event.before }}" >> $GITHUB_ENV
+          echo "to_ref=${{ github.event.after }}" >> $GITHUB_ENV
+
+      - name: Generate commit refs for push to PR
+        if: github.event_name == 'pull_request'
+        run: |
+          echo "from_ref=${{ github.base_ref }}" >> $GITHUB_ENV
+          echo "to_ref=${{ github.head_ref }}" >> $GITHUB_ENV
+
+      # below is required for the pre-commit to work properly
+      - name: Fetch commit refs
+        run: |
+          git fetch origin ${{ env.from_ref }}
+          git fetch origin ${{ env.to_ref }}
+
+      - name: Set up Python
+        id: setup-python
+        uses: actions/setup-python@v5
+        with:
+          python-version: v3.12
+
+      - name: Install pre-commit
+        run: pip install pre-commit
+
+      - name: Restore pre-commit environment from cache
+        id: cache-precommit
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.cache/pre-commit
+            committed
+          key: |
+            ${{ runner.os }}-${{ env.cache-name }}-${{ hashFiles('.pre-commit-config.yaml') }}
+
+      - name: Install pre-commit dependencies
+        if: steps.cache-precommit.outputs.cache-hit != 'true'
+        run: |
+          pre-commit install-hooks
+
+          wget -q https://github.com/crate-ci/committed/releases/download/v1.1.7/committed-v1.1.7-x86_64-unknown-linux-musl.tar.gz
+          mkdir committed
+          tar -xf committed-v1.1.7-x86_64-unknown-linux-musl.tar.gz --directory committed
```

Comment:
Isn't this the same as before?

You cache the directory, but you still download the tar and extract it every time???

Reply:
If  ` ~/.cache/pre-commit` and `committed` could be extracted from the cache this step is skipped. This is happens due to the `if: steps.cache-precommit.outputs.cache-hit != 'true'` line.

Reply:
Ah...

![Added via Giphy](https://media1.giphy.com/media/v1.Y2lkPWM5NDg3NzQzdmltbmRlbDY1bnYzZ3lvMThzOW5sM3FheXFmdjQwd3FodzBtN2h4NSZlcD12MV9naWZzX3NlYXJjaCZjdD1n/83QtfwKWdmSEo/giphy.gif)